registry/storage/ociartifactmanifesthandler.go

```diff
@@ -112,3 +122,32 @@ func (ms *ociArtifactManifestHandler) verifyArtifactManifest(ctx context.Context
 
 	return nil
 }
+
+// indexReferrers indexes the subject of the given revision in its referrers index store.
+func (ms *ociArtifactManifestHandler) indexReferrers(ctx context.Context, dm *ociartifact.DeserializedManifest, revision digest.Digest) error {
+	if dm.Subject == nil {
+		return nil
+	}
+
+	// [TODO] We can use artifact type in the link path to support filtering by artifact type
+	//  but need to consider the max path length in different os
+	subjectRevision := dm.Subject.Digest
+
+	rootPath := path.Join(referrersLinkPath(ms.repository.Named().Name()), subjectRevision.Algorithm().String(), subjectRevision.Hex())
+	referenceLinkPath := path.Join(rootPath, revision.Algorithm().String(), revision.Hex(), "link")
```

Comment:
What would be proper path considering `referrers` is a first-class item like `manifests`, `blobs`?

Reply:
How would it be impacted if we delete a repository?

Reply:
Also think about how artifact type can be supported.

Hint: Hashing will be useful for user input projection.

Reply:
Currently, only PUT path is considered. How about the DELETE path in terms of indexing referrers?

Reply:
After offline discussion, I will address these issues in later prs.